packages/realm/src/Counter.ts

```diff
@@ -0,0 +1,149 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2024 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { IllegalConstructorError, Realm, UpdateMode, assert, binding } from "./internal";
+
+const REALM = Symbol("Counter#realm");
+const OBJ = Symbol("Counter#obj");
+const COLUMN_KEY = Symbol("Counter#columnKey");
+
+/**
+ * A logical counter representation for performing numeric updates that need
+ * to be synchronized as sequentially consistent events rather than individual
+ * reassignments of the number.
+ *
+ * For instance, offline Client 1 and Client 2 which both see `Counter.value`
+ * as `0`, can both call `Counter.increment(1)`. Once online, the value will
+ * converge to `2`.
+ * @note
+ * __Not supported__:
+ *
+ * Counter types are __not__ supported as:
```

Comment:
My eyes stopped at this, since I haven't seen use use this markdown syntax elsewhere.

I wonder if it's possible / better to use an h2 here instead?

```suggestion
 * ## Not supported
 *
 * Counter types are __not__ supported as:
```

Or since this the header is sort of repeated with the pretext to the list of items, it would be equally clear (and less verbose) to just leave out the heading entirely.

```suggestion
 * Counter types are __not__ supported as:
```

Reply:
It's a bold syntax, but yeah a heading may be more suitable 👍 Tested with h3 which renders a bit better than h2 I think 🙃

Reply:
✅